### PR TITLE
Support list from resourceVersion for partitioned API servers.

### PR DESCRIPTION
**Pull request by Sindica**

### CODE REVIEW

**Comment by Sindica** [comment]
https://github.com/futurewei-cloud/arktos/issues/516

File: staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go [hunk 1]
```diff
@@ -456,6 +460,73 @@ func (c *$.type|privatePlural$) List(opts $.ListOptions|raw$) (result *$.resultT
...
+					return
+				} else {
+					continue
+				}
+			}
+
+			currentResult, _ := results[j]
+			if result.Kind == "" {
+				result.TypeMeta = currentResult.TypeMeta
+				result.ListMeta = currentResult.ListMeta
```
Comment (chenqianfzh) [inline staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go:515]: I guess you add the check "if result.kind == """ if for efficiency concern. If so, you need set the result.Kind in the block after that check.

File: staging/src/k8s.io/client-go/kubernetes/typed/apps/v1/deployment.go [hunk 2]
```diff
@@ -102,6 +103,72 @@ func (c *deployments) List(opts metav1.ListOptions) (result *v1.DeploymentList,
...
+				}
+			}
+		}
+
+		for _, item := range itemsMap {
+			result.Items = append(result.Items, *item)
+		}
+		return
+	}
+
```
Comment (chenqianfzh) [inline staging/src/k8s.io/client-go/kubernetes/typed/apps/v1/deployment.go:171]: Explain the use scenario of the following code.

Comment (Sindica) [reply staging/src/k8s.io/client-go/kubernetes/typed/apps/v1/deployment.go:183]: Done

**Pull request merged**